lib/puppet/type/notify.rb

```diff
@@ -6,7 +6,7 @@ module Puppet
   Type.newtype(:notify) do
     @doc = "Sends an arbitrary message to the agent run-time log."
 
-    newproperty(:message) do
+    newproperty(:message, :idempotent => false) do
```

Comment:
Fwiw, I'm 👍 on the use of `:idempotent` here as opposed to the whitelist approach. And thanks for that experiment @Iristyle - it seems like a reasonable proxy.